core/src/main/java/org/aerogear/mobile/core/metrics/MetricsPublisher.java

```diff
@@ -1,11 +1,76 @@
 package org.aerogear.mobile.core.metrics;
 
+import static org.aerogear.mobile.core.utils.SanityCheck.nonNull;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import android.content.Context;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 
 import org.aerogear.mobile.core.Callback;
+import org.aerogear.mobile.core.MobileCore;
+import org.aerogear.mobile.core.logging.Logger;
+import org.aerogear.mobile.core.metrics.impl.AppMetrics;
+import org.aerogear.mobile.core.metrics.impl.DeviceMetrics;
+import org.aerogear.mobile.core.utils.ClientIdGenerator;
+
+public abstract class MetricsPublisher {
+
+    private static final Logger LOGGER = MobileCore.getLogger();
+
+    private final Context context;
+    private final Metrics[] defaultMetrics;
+
+    public MetricsPublisher(final Context context) {
+        this.context = context;
+
+        defaultMetrics = new Metrics[] {new AppMetrics(context), new DeviceMetrics()};
+    }
+
+    /**
+     * Parse metrics into a JSONObject and add communs information for all metrics requests:
+     *
+     * - clientId - timestamp - type - app - device
+     *
+     * @param type the type of metrics
+     * @param metrics metrics
+     * @return a JSONObject
+     */
+    protected JSONObject parseMetrics(final String type, final Metrics[] metrics) {
```

Comment:
let's rename it to createMetricsPayload or createMetricsJSONObject.
I am not a native speaker, but "parsing" is not what happens here.